src/event_notifier.rs

```diff
@@ -0,0 +1,291 @@
+                let set_fd_flags = |fd: RawFd| -> io::Result<()> {
+                    let flags = unsafe { libc::fcntl(fd, libc::F_GETFL) };
+                    if flags < 0 {
+                        return Err(io::Error::last_os_error());
+                    }
+                    if unsafe { libc::fcntl(fd, libc::F_SETFL, flags | flag) } < 0 {
+                        return Err(io::Error::last_os_error());
+                    }
+                    Ok(())
+                };
+
+                if let Err(e) = set_fd_flags(fds[0]) {
+                    unsafe {
+                        libc::close(fds[1]);
+                    }
+                    return Err(e);
+                }
+                if let Err(e) = set_fd_flags(fds[1]) {
+                    unsafe {
+                        libc::close(fds[0]);
+                    }
+                    return Err(e);
+                }
+
+                // SAFETY: This is safe because we checked ret for success and know
+                // the kernel gave us an fd that we own.
+                Ok(EventNotifier {
+                    rfd: unsafe { File::from_raw_fd(fds[0]) },
+                    wfd: unsafe { File::from_raw_fd(fds[1]) },
+                })
+            }
+        }
+    }
+
+    /// Add a value to the eventfd's counter.
+    /// Add a value to the pipfd's buffer.
+    ///
+    /// For the eventfd, when the addition causes the counter overflow, this would
+    /// either block until a [`read`](http://man7.org/linux/man-pages/man2/read.2.html)
+    /// is performed on the file descriptor, or fail with the
+    /// error EAGAIN if the file descriptor has been made nonblocking.
+    ///
+    /// # Arguments
+    ///
+    /// * `v`: the value to be added to the eventfd's counter.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// extern crate vmm_sys_util;
+    /// use vmm_sys_util::event_notifier::EventNotifier;
+    /// #[cfg(any(target_os = "linux", target_os = "android"))]
+    /// use vmm_sys_util::event_notifier::EFD_NONBLOCK;
+    /// #[cfg(target_os = "macos")]
+    /// use vmm_sys_util::event_notifier::O_NONBLOCK;
+    ///
+    /// #[cfg(any(target_os = "linux", target_os = "android"))]
+    /// let evt = EventNotifier::new(EFD_NONBLOCK).unwrap();
+    /// #[cfg(target_os = "macos")]
+    /// let evt = EventNotifier::new(O_NONBLOCK).unwrap();
+    /// evt.write(55).unwrap();
+    /// ```
+    pub fn write(&self, v: u64) -> result::Result<(), io::Error> {
+        // SAFETY: This is safe because we made this fd and the pointer we pass
+        // can not overflow because we give the syscall's size parameter properly.
+        let ret = unsafe {
+            write(
+                self.wfd.as_raw_fd(),
+                &v as *const u64 as *const c_void,
+                mem::size_of::<u64>(),
+            )
+        };
+        if ret <= 0 {
+            Err(io::Error::last_os_error())
+        } else {
+            Ok(())
+        }
```

Comment:
I realize this is copied from eventfd.rs, but we can use the standard library for this:
```suggestion
	self.wfd.write_all(&v.to_ne_bytes())
```
So if you end up respinning this, could you also update eventfd.rs? :o